### PR TITLE
Refactor GraphQL documentation

### PR DESCRIPTION
**Pull request by peterdeme**
# Description of the change

## Checklist

Please make sure that the proposed change checks all the boxes below before requesting a review:

- [x] I have reviewed the [guidelines for contributing](https://github.com/spacelift-io/user-documentation/blob/main/CONTRIBUTING.md) to this repository.
- [x] The preview looks fine.
- [x] The tests pass.
- [x] The commit history is clean and meaningful.
- [x] The pull request is opened against the `main` branch.
- [x] The pull request is no longer marked as a draft.
- [x] You agree to license your contribution under the [MIT license](https://github.com/spacelift-io/user-documentation/blob/main/LICENSE) to Spacelift.

If the proposed change is ready to be merged, please request a review from `@spacelift-io/solutions-engineering`. Someone will review and merge the pull request.

Thank you for your contribution! 🙇

### CODE REVIEW

File: docs/integrations/api.md [hunk 1]
```diff
@@ -171,23 +201,208 @@ mutation GetSpaceliftToken($token: String!) {
...
+Here, make sure you fill the first three variables properly:
+
+- `BASE_URL` should be the URL of your Spacelift account. For example, `https://my-account.app.spacelift.io`
+- `API_KEY_ID` is the ID of the API key you created in [the previous step](#spacelift-api-key-token). It should be a 26-character [ULID](https://github.com/ulid/spec).
+- `API_KEY_SECRET` can be found in the file that was downloaded when you created the API key.
+
+Don't worry about the 4th.
+
+<p align="center">
+  <img src="../assets/screenshots/graphql-insomnia-fill-vars.png"/>
```
Comment (jmfontaine) [inline docs/integrations/api.md:379]: Please use Markdown syntax.

File: docs/integrations/api.md [hunk 2]
```diff
@@ -171,23 +201,208 @@ mutation GetSpaceliftToken($token: String!) {
...
+<p align="center">
+  <img src="../assets/screenshots/graphql-insomnia-01-import.png"/>
+</p>
+
+The `Spacelift` collection will appear. Click on it.
+
+On the top left corner, click the `🔵 Spacelift` icon, then choose `Manage Environments`.
+
+<p align="center">
+  <img src="../assets/screenshots/graphql-insomnia-02-manage-envs.png"/>
```
Comment (jmfontaine) [inline docs/integrations/api.md:367]: Please use Markdown syntax.

File: docs/integrations/api.md [hunk 3]
```diff
@@ -171,23 +201,208 @@ mutation GetSpaceliftToken($token: String!) {
...
+            "_type": "environment"
+        	}
+    	]
+	}
+	```
+
+In the home screen of Insomnia, click on `Import From` then click on `Clipboard`.
+
+<p align="center">
+  <img src="../assets/screenshots/graphql-insomnia-01-import.png"/>
```
Comment (jmfontaine) [inline docs/integrations/api.md:359]: Please use Markdown syntax.

File: docs/integrations/api.md [hunk 4]
```diff
@@ -45,25 +107,40 @@ One approach to generating this token is using the Spacelift [spacectl](https://
...
-![Create and Configure your new API Key.](../assets/screenshots/3-create-api-key.png)
+<p align="center">
+  <img src="../assets/screenshots/api-key-creation-03.png"/>
+</p>
 
 Once you click the _Add Key_ button, the API Key will be generated and a file will be automatically downloaded. The file contains the API token in two forms - one to be used with our API, and the other one as a `.terraformrc` snippet to access your [private modules](../vendors/terraform/module-registry.md) outside of Spacelift:
 
-![Download your API Key.](../assets/screenshots/4-download-api-key.png)
+<p align="center">
+  <img src="../assets/screenshots/api-key-creation-04.png"/>
```
Comment (jmfontaine) [inline docs/integrations/api.md:142]: Please use Markdown syntax.

File: docs/integrations/api.md [hunk 5]
```diff
@@ -45,25 +107,40 @@ One approach to generating this token is using the Spacelift [spacectl](https://
...
 
 The API key creation form will allow you to specify an arbitrary key name, along with the _Admin_ setting and the list of _teams_. If the key is given admin privileges, it has full access to the Spacelift API and won't be subject to [access policies](../concepts/policy/stack-access-policy.md).
 
 For non-administrative keys, you may want to add a **virtual** list of teams that the key should "belong to" so that existing access policies based on [GitHub teams](source-control/github.md#team-based-access) or [SAML assertions](single-sign-on/README.md#setting-up-the-integration) can work with your API keys just as they do with regular users.
 
 Without further ado, let's create a non-administrative API key with virtual membership in two teams: _Developers_ and _DevOps:_
 
-![Create and Configure your new API Key.](../assets/screenshots/3-create-api-key.png)
+<p align="center">
+  <img src="../assets/screenshots/api-key-creation-03.png"/>
```
Comment (jmfontaine) [inline docs/integrations/api.md:136]: Please use Markdown syntax.

File: docs/integrations/api.md [hunk 6]
```diff
@@ -4,32 +4,94 @@ description: Describes how to authenticate and use the Spacelift GraphQL API.
...
+
+As of today, Postman does not support GraphQL natively, so our choice of tool is [Insomnia](https://insomnia.rest/download). Insomnia is a free, open-source tool that allows you to easily create and manage API requests.
+
+## Usage Demo
+
+The below guide walks through an example of generating your Spacelift token with spacectl and using it to communicate with Spacelift.
+
+**Prerequisites:**
+
+- [Insomnia](https://insomnia.rest/download) downloaded and installed
```
Comment (jmfontaine) [inline docs/integrations/api.md:78]:
```suggestion
- [Insomnia](https://insomnia.rest/download){: rel="nofollow"} downloaded and installed
```

File: docs/integrations/api.md [hunk 7]
```diff
@@ -4,32 +4,94 @@ description: Describes how to authenticate and use the Spacelift GraphQL API.
...
+}
+```
+
+</details>
+
+## Recommendation
+
+Our recommendation is to use the [Spacelift API Key](#spacelift-api-key-token) to authenticate with the GraphQL API.
+
+As of today, Postman does not support GraphQL natively, so our choice of tool is [Insomnia](https://insomnia.rest/download). Insomnia is a free, open-source tool that allows you to easily create and manage API requests.
```
Comment (jmfontaine) [inline docs/integrations/api.md:70]:
```suggestion
As of today, Postman does not support GraphQL natively, so our choice of tool is [Insomnia](https://insomnia.rest/download){: rel="nofollow"}. Insomnia is a free, open-source tool that allows you to easily create and manage API requests.
```

File: docs/integrations/api.md [hunk 8]
```diff
@@ -171,23 +201,208 @@ mutation GetSpaceliftToken($token: String!) {
 
 Assuming all went well, the result of the above query will return your JWT bearer token, which you will now be able to use to authenticate other queries. Once acquired, ensure you use this bearer token in your requests. If you want to access the API reliably in an automated way, we suggest using the [Spacelift API Key > JWT Token](api.md#spacelift-api-key-greater-than-jwt-token) approach as Spacelift tokens expire after 1 hour.
 
+## Insomnia setup
+
+You can create request libraries in Insomnia to make it easier to work with the Spacelift API. You can also automate the JWT token generation process using the [Environment Variables](https://docs.insomnia.rest/insomnia/environment-variables) feature.
```
Comment (jmfontaine) [inline docs/integrations/api.md:206]:
```suggestion
You can create request libraries in Insomnia to make it easier to work with the Spacelift API. You can also automate the JWT token generation process using the [Environment Variables](https://docs.insomnia.rest/insomnia/environment-variables){: rel="nofollow"} feature.
```

File: docs/integrations/api.md [hunk 9]
```diff
@@ -171,23 +201,208 @@ mutation GetSpaceliftToken($token: String!) {
...
+On the top left corner, click the `🔵 Spacelift` icon, then choose `Manage Environments`.
+
+<p align="center">
+  <img src="../assets/screenshots/graphql-insomnia-02-manage-envs.png"/>
+</p>
+
+Here, make sure you fill the first three variables properly:
+
+- `BASE_URL` should be the URL of your Spacelift account. For example, `https://my-account.app.spacelift.io`
+- `API_KEY_ID` is the ID of the API key you created in [the previous step](#spacelift-api-key-token). It should be a 26-character [ULID](https://github.com/ulid/spec).
```
Comment (jmfontaine) [inline docs/integrations/api.md:373]:
```suggestion
- `API_KEY_ID` is the ID of the API key you created in [the previous step](#spacelift-api-key-token). It should be a 26-character [ULID](https://github.com/ulid/spec){: rel="nofollow"}.
```

File: docs/integrations/api.md [hunk 10]
```diff
@@ -4,32 +4,94 @@ description: Describes how to authenticate and use the Spacelift GraphQL API.
...
-- Insomnia downloaded and installed
+<details>
+<summary>An example of request and response</summary>
+
+```
+$ curl --request POST \
+  --url http://<account-name>.app.spacelift.io/graphql \
+  --header 'Authorization: Bearer <token>' \
+  --header 'Content-Type: application/json' \
+  --data '{"query":"{ \n\tstacks\n\t{\n\t\tid\n\t\tname,\n\t\tadministrative,\n\t\tcreatedAt,\n\t\tdescription\n\t}\n}"}'
```
Comment (jmfontaine) [inline docs/integrations/api.md:21]: Do we need all the `\n` and `\t`?

File: docs/integrations/api.md [hunk 11]
```diff
@@ -4,32 +4,94 @@ description: Describes how to authenticate and use the Spacelift GraphQL API.
...
+All requests need to be authenticated using a [JWT](https://jwt.io/) bearer token, which we will discuss in more detail below.
 
-In order to use the API, you will need a **bearer token** to authenticate your requests. There are currently three ways of obtaining this token:
+There are currently three ways of obtaining this token:
 
-1. [SpaceCTL CLI > Token](api.md#spacectl-cli-greater-than-token) (easiest method)
-2. [Spacelift API Key > Token](api.md#spacelift-api-key-greater-than-token)
-3. [Personal GitHub Token > Token](api.md#personal-github-token-greater-than-token)
+1. [SpaceCTL CLI > Token](#spacectl-cli-token) - for temporary usage
+2. [Spacelift API Key > Token](#spacelift-api-key-token) - for long-term usage (**recommended**)
```
Comment (jmfontaine) [inline docs/integrations/api.md:93]: Should this not be the first item if it is the new recommended way?

File: docs/integrations/api.md (hunk 1 above)
Comment (peterdeme) [reply docs/integrations/api.md:379]: Markdown syntax doesn't support centering, and pictures look weird without it.

Comment (jmfontaine) [reply docs/integrations/api.md:379]: I see bigger problems with the pictures (inconsistency in the arrows color and some are like zoomed-in) than positioning but I can live with this syntax. 😄

**Comment by peterdeme** [comment]
@jmfontaine I'll refactor this to use Postman, apprently it supports graphql now. Insomnia is trash so I'm happy

File: docs/integrations/api.md (hunk 1 above)
Comment (peterdeme) [reply docs/integrations/api.md:379]: okay, I'll fix the screenshots

**Comment by peterdeme** [comment]
@jmfontaine good point. actually, I won't add Postman now because I've spent already too much on this :D I'll just create new screenshots and call it a day

File: docs/integrations/api.md (hunk 1 above)
Comment (peterdeme) [reply docs/integrations/api.md:380]: updated to black arrows

Comment (jmfontaine) [reply docs/integrations/api.md:380]: The black arrows were the inconsistency. We use red arrows everywhere in the documentation because they stand out. 😄

Comment (peterdeme) [reply docs/integrations/api.md:380]: ahh ok, I'll fix it

Comment (peterdeme) [reply docs/integrations/api.md:380]: @jmfontaine fixed it

**Review by jmfontaine — COMMENTED** [review]
@peterdeme I am sorry but there is one last screenshot with a black arrow. 😄 

Otherwise, it looks good to me.

**Comment by peterdeme** [comment]
@jmfontaine I specifically looked at that screenshot and was like "oh this one doesn't have arrows, cool".

I didn't notice it because it was black 😂 fixed

**Comment by jmfontaine** [comment]
> I didn't notice it because it was black

@peterdeme That was exactly my point. 🤣

**Comment by jmfontaine** [comment]
@peterdeme I tried in two different browsers and `docs/assets/screenshots/api-key-ulid.png` still shows a black arrow. Am I missing something?

CleanShot 2023-01-24 at 10 34 25

**Comment by peterdeme** [comment]
@jmfontaine ahh, no I missed it. pushed a new.

**Review by jmfontaine — APPROVED** [review]
🥳

**Pull request merged**